diesel/src/query_dsl/join_dsl.rs

```diff
@@ -44,34 +44,35 @@ where
     }
 }
 
+/// Specify the `ON` clause for a join statement. This will override
+/// any implicit `ON` clause that would come from [`joinable!`]
+///
+/// [`joinable!`]: ../macro.joinable.html
+///
+/// # Example
+///
+/// ```rust
+/// # #[macro_use] extern crate diesel;
+/// # include!("../doctest_setup.rs");
+/// # use schema::{users, posts};
+/// #
+/// # fn main() {
+/// #     let connection = establish_connection();
+/// let data = users::table
+///     .left_join(posts::table.on(
+///         users::id.eq(posts::user_id).and(
+///             posts::title.eq("My first post"))
```

Comment:
Btw, I'd format this as

```rust
users::id.eq(posts::user_id)
.and(posts::title.eq("My first post"))
```

Reply:
This is just moved from the other location. I wonder if we can set up a way to run our docs code through rustfmt.